src/Diagnostics.DataProviders/DataProviders/ISupportObserverDataProvider.cs

```diff
@@ -0,0 +1,32 @@
+﻿using Diagnostics.ModelsAndUtils.Models;
+using Newtonsoft.Json.Linq;
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Diagnostics.DataProviders
+{
+    public interface ISupportObserverDataProvider
+    {
+        Task<dynamic> GetSite(string siteName);
+        Task<dynamic> GetSite(string stampName, string siteName);
+        Task<IEnumerable<string>> GetSiteHostNames(string siteName);
```

Comment:
Can you add the getsitehostnames(siteName, stampName) for which there is an open bug about? Or just combine them in the first go